### PR TITLE
cleanups

### PR DESCRIPTION
**Pull request by SpacePossum**
bits and bites 
some stuff should go to lower branches, however I like to see the CI first

expect at least one failure related to `PHPUnitGoodPractices` which I hope to patch @ that repo

### CODE REVIEW

File: tests/Test/AbstractIntegrationTestCase.php [hunk 1]
```diff
@@ -147,12 +148,13 @@ public function provideIntegrationCases()
...
         $tests = [];
 
+        /** @var SplFileInfo $file */
         foreach (Finder::create()->files()->in($fixturesDir) as $file) {
             if ('test' !== $file->getExtension()) {
                 continue;
             }
 
-            $tests[] = [
+            $tests[$file->getPathname()] = [
```
Comment (SpacePossum) [inline tests/Test/AbstractIntegrationTestCase.php]: sample of why this change

old:
```
 9. 104ms to run PhpCsFixer\Tests\IntegrationTest:testIntegration with data set #145
 10. 103ms to run PhpCsFixer\Tests\IntegrationTest:testIntegration with data set #137
```

new:

```
 6. 160ms to run PhpCsFixer\Tests\IntegrationTest:testIntegration with data set "/home/possum/work/PHP-CS-Fixer/tests/Fixtures/Integration/misc/PHP7.test"
 7. 114ms to run PhpCsFixer\Tests\IntegrationTest:testIntegration with data set "/home/possum/work/PHP-CS-Fixer/tests/Fixtures/Integration/set/@Symfony_whitespaces.test"
 8. 104ms to run PhpCsFixer\Tests\IntegrationTest:testIntegration with data set "/home/possum/work/PHP-CS-Fixer/tests/Fixtures/Integration/set/@Symfony.test"
```

File: tests/AutoReview/FixerFactoryTest.php [hunk 2]
```diff
@@ -152,7 +151,6 @@ public function provideFixersPriorityCases()
             [$fixers['php_unit_fqcn_annotation'], $fixers['php_unit_ordered_covers']],
             [$fixers['php_unit_no_expectation_annotation'], $fixers['no_empty_phpdoc']],
             [$fixers['php_unit_no_expectation_annotation'], $fixers['php_unit_expectation']],
-            [$fixers['php_unit_strict'], $fixers['php_unit_construct']],
```
Comment (SpacePossum) [inline tests/AutoReview/FixerFactoryTest.php]: there is no prio issue here, both fixers deal with `assertEquals` and `assertSame` and v.v.

File: tests/AutoReview/FixerFactoryTest.php [hunk 3]
```diff
@@ -264,13 +262,10 @@ public function testFixersPriorityPairsHaveIntegrationTest(FixerInterface $first
             'class_attributes_separation,braces.test',
             'class_attributes_separation,indentation_type.test',
             'indentation_type,phpdoc_indent.test',
-            'line_ending,single_blank_line_at_eof.test',
```
Comment (SpacePossum) [inline tests/AutoReview/FixerFactoryTest.php]: there is prio issue (anymore) as both fixers use the same line break configuration

File: tests/Fixtures/Integration/priority/no_empty_statement,multiline_whitespace_before_semicolons.test [hunk 4]
```diff
@@ -0,0 +1,21 @@
+--TEST--
```
Comment (keradus) [inline tests/Fixtures/Integration/priority/no_empty_statement,multiline_whitespace_before_semicolons.test]: duplicates https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3436/files

Comment (SpacePossum) [reply tests/Fixtures/Integration/priority/no_empty_statement,multiline_whitespace_before_semicolons.test]: ^_^
:+1: 
I remove it later, I was just going through the missing ones in arbitrary order without checking open PR's :(
